### PR TITLE
airshipper: 0.7.0 -> 0.10.0

### PR DESCRIPTION
**Pull request by binarydigitz01**
…lob/v0.10.0/CHANGELOG.md

###### Description of changes

Changelog: https://gitlab.com/veloren/airshipper/-/blob/v0.10.0/CHANGELOG.md

###### Things done

Some assets files were changed, so I needed to change them, and add compilation support for rust unstable features.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by binarydigitz01** [comment]
Once this PR is merged I would also make a backport for nixos-unstable, as it seems airshipper hasn't been updated for quite a while.

**Comment by figsoda** [comment]
backporting nixos-unstable is not needed since things from master automatically get merged into that channel

If you meant to say the `release-` branches, feel free to do so

File: pkgs/games/airshipper/default.nix [hunk 1]
```diff
@@ -68,7 +63,7 @@ rustPlatform.buildRustPackage rec {
         libXcursor
       ];
     in
-    ''
+      ''
```
Comment (figsoda) [inline pkgs/games/airshipper/default.nix:66]: this doesn't seem right
```suggestion
    ''
```

Comment (binarydigitz01) [reply pkgs/games/airshipper/default.nix:66]: You are right, indentation problems on my part

**Review by figsoda — APPROVED** [review]
changes lgtm

just making sure, did you want this to be backported to 22.11?

**Review by figsoda — CHANGES_REQUESTED** [review]
I was getting errors when I was trying to install veloren, we might need to package the patcher as well: https://github.com/veloren/Airshipper/blob/636405c81aee19d1103a6973b16922ac60804b01/flake.nix#L107

I will hand this to @yusdacra

**Comment by binarydigitz01** [comment]
Nope, you're right it doesn't work with the new launcher.

**Comment by yusdacra** [comment]
We need to add the patcher here since Airshipper doesn't bundle a Nix expression for patching anymore (better for us). You can just copy the patcher from the flake and use it here I believe (https://gitlab.com/veloren/airshipper/-/blob/master/flake.nix#L40-57).

**Comment by binarydigitz01** [comment]
@yusdacra I have tried to add the patcher, but I was getting the error: ```patchelf: not an ELF executable```
Could you please take a look?

**Comment by binarydigitz01** [comment]
Well now i don't get an error, but for some reason the veloren installation still fails. But there are no errors, here are the logs:
```❯ ./result/bin/airshipper 
2022-12-07T19:58:09.377356Z  INFO airshipper::logger: Setup terminal and file logging. path="/home/binarydigitz01/.local/share/airshipper" file="airshipper.log"
2022-12-07T19:58:09.377401Z  INFO airshipper::cli: Visit https://book.veloren.net/ for an FAQ and Troubleshooting
2022-12-07T19:58:09.606396Z  WARN wgpu_hal::gles::egl: Re-initializing Gles context due to Wayland window    
2022-12-07T19:58:09.633612Z  WARN wgpu_hal::gles::adapter: Detected skylake derivative running on mesa i915. Clears to srgb textures will use manual shader clears.    
2022-12-07T19:58:09.640556Z  WARN wgpu_hal::gles::egl: Re-initializing Gles context due to Wayland window    
2022-12-07T19:58:11.287903Z ERROR airshipper::error: Error while performing network operations. => error sending request for url (https://veloren.net/rss.xml): connection error: unexpected end of file
2022-12-07T19:58:11.287904Z ERROR airshipper::error: Error while performing network operations. => error sending request for url (https://veloren.net/community-showcase/rss.xml): connection error: unexpected end of file
2022-12-07T19:58:11.288490Z ERROR airshipper::gui::rss_feed: Failed to fetch RSS feed e=NetworkError
2022-12-07T19:58:11.288537Z ERROR airshipper::gui::rss_feed: Failed to fetch RSS feed e=NetworkError
2022-12-07T19:59:23.817393Z  INFO airshipper::io::fs: Unzipping to "/home/binarydigitz01/.local/share/airshipper/profiles/default"
2022-12-07T19:59:30.431286Z  INFO airshipper::nix: Patching voxygen and server CLI executable files for NixOS
2022-12-07T19:59:30.431300Z  INFO airshipper::nix: Executing "/nix/store/sivchbyrb2pmlzvai5kq27aa95zr75c6-patch" on directory "/home/binarydigitz01/.local/share/airshipper/profiles/default"
```

**Comment by binarydigitz01** [comment]
I have applied a patch sent by yusducra, and now installing veloren works. I think that this is branch can be merged.
@figsoda @yusdacra

**Comment by yusdacra** [comment]
Result of `nixpkgs-review pr 204725` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>airshipper</li>
  </ul>
</details>

**Review by 90-008 — APPROVED** [review]

**Review by figsoda — APPROVED** [review]

**Pull request merged**